libs/components/ag-grid/src/lib/modules/ag-grid/ag-grid-wrapper.component.ts

```diff
@@ -287,34 +287,41 @@ export class SkyAgGridWrapperComponent
     }
   }
 
-  #updateGridTheme(themeSettings?: SkyThemeSettings): void {
-    let agTheme: 'default' | 'modern-light' | 'modern-dark';
-    if (themeSettings?.theme.name === 'modern') {
-      agTheme = `modern-${themeSettings.mode.name}` as
-        | 'modern-light'
-        | 'modern-dark';
-    } else {
-      agTheme = `default`;
-    }
-    const variation = this.#hasEditableClass ? 'data-entry-grid' : 'data-grid';
+  #updateGridTheme(
+    hasEditableClass: boolean,
+    themeSettings?: SkyThemeSettings,
+  ): void {
+    const agTheme = agGridTheme(hasEditableClass, themeSettings, this.compact);
+    const previousValue = this.#wrapperClasses.getValue();
+    const previousTheme = previousValue.find((c) => c.startsWith('ag-theme-'));
     let value = [
-      ...this.#wrapperClasses
-        .getValue()
-        .filter((c) => !c.startsWith('ag-theme-')),
-      `ag-theme-sky-${variation}-${agTheme}`,
+      ...previousValue.filter((c) => !c.startsWith('ag-theme-')),
```

Comment:
is it possible that there is more than one theme class? If not, can we just use the `previousTheme` we already found in this list?

Reply:
This is the exclusion of the previous theme.